railties/lib/rails/generators/rails/app/templates/config/puma.rb.tt

```diff
@@ -2,38 +2,53 @@
 # are invoked here are part of Puma's configuration DSL. For more information
 # about methods provided by the DSL, see https://puma.io/puma/Puma/DSL.html.
 
-# Puma can serve each request in a thread from an internal thread pool.
-# The `threads` method setting takes two numbers: a minimum and maximum.
-# Any libraries that use thread pools should be configured to match
-# the maximum value specified for Puma. Default is set to 5 threads for minimum
-# and maximum; this matches the default thread size of Active Record.
-max_threads_count = ENV.fetch("RAILS_MAX_THREADS") { 5 }
-min_threads_count = ENV.fetch("RAILS_MIN_THREADS") { max_threads_count }
-threads min_threads_count, max_threads_count
-
-# Specifies that the worker count should equal the number of processors in production.
-if ENV["RAILS_ENV"] == "production"
+rails_env = ENV.fetch("RAILS_ENV", "development")
+
+# Puma starts a configurable number of processes (workers) and each process
+# serves each request in a thread from an internal thread pool.
+#
+# As a rule of thumb, increasing the number of threads will increase how much
+# traffic a given process can handle (throughput), but due to Ruby's
+# Global VM Lock (GVL) it has diminishing returns, and will degrade the
+# response time (latency) of the application.
+#
+# The ideal number of threads per worker depends both on how much time the
```

Comment:
I'd just lay out the tradeoff:

"Increasing threads per worker increases throughput for applications which spend significant time waiting on I/O, but can also increase latency. Decreasing threads per worker keeps latency consistent and low for all applications, at the cost of maximum throughput."

Reply:
It seems like it's very similar to the "rule of thumb" I describe above.